code-challenges.test.js

```diff
@@ -0,0 +1,126 @@
+// ASSESSMENT 3: Coding Practical Questions with Jest
+
+// Please read all questions thoroughly
+// Pseudo coding is REQUIRED
+// If you get stuck, please leave comments to help us understand your thought process
+
+// Use test driven development to complete the following questions
+// Add appropriate dependencies: $ yarn add jest
+
+// Reminder: The test will call your function
+// Run the file with the following command: $ yarn jest
+
+// --------------------1) Create a function that takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+// describe ("fibonacci", () => {
+//   it("returns an array the same amount of digits that the var number is. But the new array is the fibonaci sequence", () => {
+//     expect(fibonacci(fibLength1)).toEqual([0, 1, 1, 2, 3, 5, 8])
+//     expect(fibonacci(fibLength2)).toEqual([0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+//   })
+// })
+
+
+// const fibLength1 = 7
+// Expected output: [0, 1, 1, 2, 3, 5, 8]
+
+// const fibLength2 = 11
+// Expected output: [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+
+// b) Create the function that makes the test pass.
+
+// const fibonacci = n =>
+//   Array.from({ length : n }).reduce(
+//     (acc, val, i) => acc.concat(i > 1 ? acc[i - 1] + acc[i - 2] : i),
+//     []
+//   );
+// console.log(fibonacci(fibLength1)); 
+// console.log(fibonacci(fibLength2)); 
+
+// Pseudo code: create a variable, turn the var into an array with .from() round the numbers with .reduce. Add in our for loop and console log the the function and var.
+// I tried to get rid of the 0 too but couldn't manage.
```

Comment:
This is the issue with finding code online and trying to use it without understanding what is going on. There is too much going on here to explain without overwhelming you with the logic. Your pseudo code also shows that you do not know what is being executed in this code. Keep in mind that it is much better for you in the long run to try to write your own code solution and not get far vs finding something to paste and trying to make it work. If you want to go over this approach, bring it up during assessment review and I would be happy to go through it 🙂